doc/design/argo_workflow_on_sqlflow.md

```diff
@@ -81,16 +81,35 @@ We implement the `Fetch` API as the following pseudo-code:
 ``` go
 func Fetch(req *FetchRequest) *FetchResponse {
   // kubectl get pod ...
-  pod := getPod(req)
+  step := getStep(req)
 
   // kubectl get workflow ...
   wf := getWorkflow(req)
 
-  logs, newLogOffset, isFinished := fetchPodLogs(pod, req.logOffset)
-
-  newReq := newFetchRequest(wf, req.stepId, newLogOffset, isFinished)
+  // return step phase logs if the phase changed
+  stepPhaseLogs := fetchStepPhaseLogs(wf, step)
+
+  // append step phase status logs to Response
+  responses := []pb.Response{&Message{stepPhaseLogs}}
+
+  eof := false // True for end of job
+
+  if podComplete(pod) {
+    // if the current step execute a standard SQL, the query result would be output
+    // as protobuf message string format with prefix SQLFLOW_PROTOBUF:
```

Comment:
is the serialized message using `pbtxt`?

Reply:
The comment used `Unmarshl` instead of `UnmarshlText`, I can do more tests when I implement it.